python/cudf/cudf/core/frame.py

```diff
@@ -1448,7 +1448,7 @@ def searchsorted(
 
         # Return result as cupy array if the values is non-scalar
         # If values is scalar, result is expected to be scalar.
-        result = cupy.asarray(outcol.data_array_view)
+        result = cupy.asarray(outcol._data_array_view)
```

Comment:
So this is fine to not copy, because cupy is taking ownership of the data and column goes out of scope.

Reply:
Right

Reply:
That's correct.